### ISSUE TITLE
How to help with MPC-HC translations

### ISSUE THREAD

**Issue by clsid2** [clsid2/mpc-hc#3138]
If you want to help complete/update/improve a translation of MPC-HC, then you can do that by singing up here:
https://explore.transifex.com/mpchc/mpchc/

I am not going to add any new languages. Certainly not uncommon exotic languages.

Click here:
![image](https://user-images.githubusercontent.com/3324395/71769509-447b5080-2ed7-11ea-8485-b630396cb0a7.png)

@adipose manages the MPC-HC project on Transifex. If it takes longer than a week to get your account approved, then drop a message below.

Translation updates get merged from Transifex to here on Github once every few months.

**Comment by Amirhan-Taipovjan-Greatest-I** [comment]
Hello!
I've applied to Tatar translating. Please, can Manager approve Me?

**Comment by adipose** [comment]
> Hello! I've applied to Tatar translating. Can Manager approve Me?

Done

**Comment by hevesij** [comment]
Hi @adipose ,
I applied for Hungarian translation a few days ago. Could you approve my request for translation? Not urgent.
Thank you!

**Comment by halbaast** [comment]
Hi @adipose, can my request for Central Kurdish (ckb) be approved please? Thanks!

**Comment by adipose** [comment]
Will check later on those requests.

**Comment by adipose** [comment]
> Hi @adipose, can my request for Central Kurdish (ckb) be approved please? Thanks!

I don't see that one, but probably because it's not currently a supported language.  Here is a commit for the last time I added a new language:

https://github.com/clsid2/mpc-hc/commit/351787f7ef909fac1ba7f6a74d25cb7ddd8aea6e

You can follow this template for everything except the .po files, and submit a PR.

**Comment by hevesij** [comment]
When will the translations from Transifex that are already 100% ready be added to the MPC-HC program? There is a translation that has been 100% for a few months, but it is still not in version 2.4.0.

**Comment by Kabelkrant** [comment]
Hello, Dutch translations have been updated and are 100% again

**Comment by adipose** [comment]
> I've sent a request to join the Ukrainian group and help finishing the translation, can you approve me please?

Done

**Comment by ghost** [comment]
https://www.transifex.com/mpchc/mpchc/ this link not working. update please
when I click on it, it redirects to 404
<details><summary>Screenshot</summary>

![Image](https://github.com/user-attachments/assets/1f9ea8d4-ecff-40db-9925-2cd29597b04a)

</details>

**Comment by adipose** [comment]
> https://www.transifex.com/mpchc/mpchc/ this link not working. update please
> when I click on it, it redirects to 404
> <details><summary>Screenshot</summary>
> 
> ![Image](https://github.com/user-attachments/assets/1f9ea8d4-ecff-40db-9925-2cd29597b04a)
> 
> </details>
> 
> 

https://app.transifex.com/mpchc/mpchc/dashboard/

**Comment by tsubasanouta** [comment]
Whenever it's convenient, could you update the Transifex file with the latest strings for the new toolbar?

**Comment by tsubasanouta** [comment]
I updated the Japanese translations to make them more consistent for the new toolbar. Please add them to MPC-HC when you have time. Thank you so much.

**Comment by tsubasanouta** [comment]
The following string is not currently in use.
DS_PPAGEADVANCED_DEFAULTTOOLBARSIZE
"Size in pixels of the default toolbar."

**Comment by tsubasanouta** [comment]
Would you see #3522?
Would you mind syncing the PO files on Transifex when you have some time?

**Comment by adipose** [comment]
I will have to work on it. Currently it has merge conflicts so I will have to see what happened.

**Comment by ghost** [comment]
Hi. Please approve Javanese request. Thanks!

**Comment by tsubasanouta** [comment]
New toolbar alignment settings have been added, and the favorites section has been redesigned. Please update the translation files on Transifex when you have time.

**Comment by adipose** [comment]
@clsid2 In the latest merge I have included an AI translation of punjabi.  This is an experiment I've put some time into.  It involves using a consensus translation from other translated languages, doing back translations to confirm accuracy, and analyzing the use of strings throughout mpc-hc.  I attached the analysis that I did (really claude did, with a lot of hand-holding to understand things correctly).

[string_usage_complete.csv](https://github.com/user-attachments/files/24227687/string_usage_complete.csv)

This is part of the project.  It might be useful to include (once fully vetted) in the pot file for reference when translating.

Punjabi was chosen as a test language because it was only 50% translated and was abandoned 6 years ago. Perhaps we can find a user to confirm it is a decent result.  If this attempt has any value I can look at updating other languages that are unmaintained.

**Comment by clsid2** [comment]
DirectShow is a name of a technology. It is not translatable.

I am not a big fan of translations. They often are too literal and many technical terms don't have proper translations in many languages, or are super long. If there are no users that can help make/correct a translation, then a language does not deserve to be included. Maybe once AI gets more mature it can be trusted more. My experience with AI is that it still returns a lot of nonsense because it is trained on Internet data that is full of nonsense and incorrect information.

But maybe another request for translator on next release notes will help.

**Comment by adipose** [comment]
>Maybe once AI gets more mature it can be trusted more. My experience with AI is that it still returns a lot of nonsense because it is trained on Internet data that is full of nonsense and incorrect information.

Yes I have experienced the same, but I have a lot of experience with working with LLMs to extract useful content.  Let me give you an example of the logic used after a lot of work.

1. **Analyze all reference languages** - Look for semantic patterns
2. **Identify consensus** - When 6+ languages agree on meaning, confidence is high
3. **Check linguistic family** - Pay special attention to related languages
4. **Verify technical terms** - See how each language handles technical vocabulary
5. **Back-translate** - Mentally translate back to English to verify accuracy

### 6. IDS_PPAGE_CAPTURE_FG0
**Original English:** "Never"

**Punjabi:** ਕਦੇ ਨਹੀਂ

**Reverse Translation:** "Never"

**Cross-Language Validation:**
- Japanese: "なし" → "None/Never"
- Korean: "없음" → "None"
- Chinese: "从不" → "Never"
- German: "Nie" → "Never"
- Spanish: "Nunca" → "Never"

**Semantic Match:** ✅ **EXACT** - Perfect literal match across all languages including Punjabi.

---

### 7. IDS_PPAGE_CAPTURE_FG1
**Original English:** "Only when switching different video types"

**Punjabi:** ਸਿਰਫ਼ ਵੱਖਰੀਆਂ ਵਿਡੀਓ ਕਿਸਮਾਂ ਸਵਿੱਚ ਕਰਨ ਵੇਲੇ

**Reverse Translation:** "Only when switching different video types"

**Cross-Language Validation:**
- Japanese: "異なる種類の動画に切り替える場合のみ" → "Only when switching to different types of videos"
- Korean: "다른 비디오 유형을 전환할 때만" → "Only when switching different video types"
- Chinese: "只有当切换不同视频类型时" → "Only when switching different video types"

**Semantic Match:** ✅ **EXACT** - All languages including Punjabi convey identical conditional timing concept.

**Technical Terms:** ✅ "video" (ਵਿਡੀਓ) preserved

**Comment by DrAhmadF** [comment]
Hello! I've applied to Arabic translating. Can Manager approve Me?

**Comment by MD-78** [comment]
Applied for German translation. Waiting for join approval.

**Comment by tsubasanouta** [comment]
Will you add "Toggle Override All Styles" and "&Sort" strings to PO files on Transifex?

**Comment by MD-78** [comment]
Updated German translation (added missing translations, fixed incorrect translations, shortened truncated translations).

**Comment by evd316** [comment]
@adipose Hi, I've submitted a request to join the Ukrainian localization team and would like to help with translation. Could you approve me?

**Comment by adipose** [comment]
> @adipose 
> Could you please update the MPC-HC PO language files on Transifex when you have a moment?”

Try https://github.com/adipose/mpc-hc-translations/releases/download/build-6054413/translation-studio-bundle-60544135.zip

**Comment by tsubasanouta** [comment]
@adipose
Could you please sync MPC-HC PO language files on Transifex when you have time?”

**Comment by xLn2** [comment]
Possible issues with latest sync:
| Translation Variable | How is look like | 
| :--- | :--- |
| IDS_PPAGEADVANCED_LOGGER | <img width="702" height="276" alt="Image" src="https://github.com/user-attachments/assets/189ef458-f94d-43c6-8886-161697fb096c" /> |
| IDS_PPAGEADVANCED_HISTORY_IN_APPDATA | <img width="711" height="93" alt="Image" src="https://github.com/user-attachments/assets/98dce738-f237-42c3-b9f2-d412310fadae" /> |

**Comment by adipose** [comment]
Those are false positives.  Transifex sees the Windows env var in `%appdata%\MPC-HC` and treats %a as a printf placeholder.  Keep the path exactly as in the English text, which you did, so both strings are fine and will go upstream with the next sync.  I also made the sync check ignore %name% env vars, so a case difference like %AppData% can't get a translation rejected.
